src/ORM/Behavior/TreeBehavior.php

```diff
@@ -224,12 +224,16 @@ public function beforeDelete(Event $event, EntityInterface $entity)
         $diff = $right - $left + 1;
 
         if ($diff > 2) {
-            $this->_table->deleteAll(function ($exp) use ($config, $left, $right) {
-                /* @var \Cake\Database\Expression\QueryExpression $exp */
-                return $exp
-                    ->gte($config['leftField'], $left + 1)
-                    ->lte($config['leftField'], $right - 1);
-            });
+            /* @var \Cake\Database\Expression\QueryExpression $expression */
+            $expression = $this->_scope($this->_table->find())
+                ->where(function ($exp) use ($config, $left, $right) {
+                    return $exp
+                        ->gte($config['leftField'], $left + 1)
+                        ->lte($config['leftField'], $right - 1);
+                })
+                ->clause('where');
+
+            $this->_table->deleteAll($expression);
```

Comment:
Instead of picking off the where clause you might want to use a subquery. The default find method _could_ add conditions from associations which will fail with this approach.

Reply:
I pushed up a commit with the changes I was proposing and a test case for this issue.